### PR TITLE
Standardize SSL settings 

### PR DESCRIPTION
**Pull request by edmocosta**
### What this PR does?
**Added the following SSL settings**
- `ssl_enabled`: Enable/disable SSL
- `ssl_keystore_path`: JKS keystore path
- `ssl_keystore_password`: JKS keystore password
- `ssl_client_authentication`: Controls the server’s behavior in regard to requesting a certificate from client connections
   
**Reviewed and deprecated the following SSL settings to comply with Logstash's [naming convention](https://github.com/elastic/logstash/issues/14905)**:
   - `ssl` in favor of `ssl_enabled`
   - `ssl_verify_mode` in favor of `ssl_client_authentication`
   - `keystore` in favor of `ssl_keystore_path`
   - `keystore_password` in favor of `ssl_keystore_password`

**Other changes**

   - Added tests for new and existing `SslSimpleBuilder` methods.
   - Moved SSL context builder logic from [SslHandlerProvider](https://github.com/logstash-plugins/logstash-input-http/pull/165/files#diff-0b9ada73a698929f27ef1e469757fcfd6158cd526b2f4d2627ce66e3f886998d) to [SslSimpleBuilder](https://github.com/logstash-plugins/logstash-input-http/pull/165/files#diff-cd477b6c3d223bfe35f4076bec10e6240dbfaf88d5d714c91e90e80e93a93700)

The behavior standardization across plugins, such as the accepted certificate formats, default values, etc will be tackled in future PRs.

---
Closes https://github.com/elastic/logstash/issues/14928

### CODE REVIEW

**Review by andsel — CHANGES_REQUESTED** [review]
Left some comments to improve, but we are quite close to the finish.

File: src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java [hunk 1]
```diff
@@ -24,6 +25,22 @@
...
+
+        SslClientVerifyMode(ClientAuth clientAuth) {
+            this.clientAuth = clientAuth;
+        }
+
+        public ClientAuth toClientAuth() {
+            return clientAuth;
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (edmocosta) [reply src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java:43]: That would be nice! Both plugins share a lot of code actually (and requirements). We've started a spike to build the foundation for a common library (https://github.com/elastic/logstash/issues/14973). After building it, the next step would be creating mixing(s) to handle all the SSL settings/clients the plugins currently support.

File: src/test/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            assertFalse(sslEngine.getNeedClientAuth());
+            assertFalse(sslEngine.getWantClientAuth());
+        } else if (SslSimpleBuilder.getVerifyMode() == SslClientVerifyMode.OPTIONAL) {
+            assertFalse(sslEngine.getNeedClientAuth());
+            assertTrue(sslEngine.getWantClientAuth());
+        } else {
+            assertTrue(sslEngine.getNeedClientAuth());
+            assertFalse(sslEngine.getWantClientAuth());
+        }
+    }
```
[the remark this answers is not in the document]
Comment (andsel) [reply src/test/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilderTest.java:245]: In the `testBuildContextWithClientAuthentication` it's using 2 times, so in that case wouldn't be ideal to identify what broke, unless we analyze the stack trace.
I think that the method is an implicit form of `if`s in test code, if we inline the helper method in the tests, we have `if`s in test code.

Hypothetically using the refactoring proposed the test code becomes more readable:
```java
    @Test
    void testBuildContextWithClientAuthentication() throws Exception {
        assertSslSimpleBuilderBuildContextWhenModeIsRequired(createSslSimpleBuilder()
                .setClientAuthentication(SslClientVerifyMode.REQUIRED, new String[]{CA})
        );

        assertSslSimpleBuilderBuildContextWhenModeIsOptional(createSslSimpleBuilder()
                .setClientAuthentication(SslClientVerifyMode.OPTIONAL, new String[]{CA})
        );
    }

    @Test
    void testBuildContextWithNoClientAuthentication() throws Exception {
        SslSimpleBuilder sslSimpleBuilder = createSslSimpleBuilder()
                .setClientAuthentication(SslClientVerifyMode.NONE, new String[]{CA});

        assertSslSimpleBuilderBuildContextWhenModeIsNone(sslSimpleBuilder);
    }
```


```java
    private void assertSslSimpleBuilderBuildContextWhenModeIsNone(SslSimpleBuilder sslSimpleBuilder) throws Exception {
        final SSLEngine sslEngine = createSSLEngineFromBuilder(sslSimpleBuilder);

        assertEquals(SslClientVerifyMode.NONE, sslSimpleBuilder.getVerifyMode());

        assertFalse(sslEngine.getNeedClientAuth());
        assertFalse(sslEngine.getWantClientAuth());
    }


    private void assertSslSimpleBuilderBuildContextWhenModeIsOptional(SslSimpleBuilder sslSimpleBuilder) throws Exception {
        final SSLEngine sslEngine = createSSLEngineFromBuilder(sslSimpleBuilder);

        assertEquals(SslClientVerifyMode.OPTIONAL, sslSimpleBuilder.getVerifyMode());

        assertFalse(sslEngine.getNeedClientAuth());
        assertTrue(sslEngine.getWantClientAuth());
    }

    private void assertSslSimpleBuilderBuildContextWhenModeIsRequired(SslSimpleBuilder sslSimpleBuilder) throws Exception {
        final SSLEngine sslEngine = createSSLEngineFromBuilder(sslSimpleBuilder);

        assertEquals(SslClientVerifyMode.REQUIRED, sslSimpleBuilder.getVerifyMode());

        assertTrue(sslEngine.getNeedClientAuth());
        assertFalse(sslEngine.getWantClientAuth());
    }


    private SSLEngine createSSLEngineFromBuilder(SslSimpleBuilder sslSimpleBuilder) throws Exception {
        final SslContext context = sslSimpleBuilder.build();

        assertTrue(context.isServer());

        final SSLEngine sslEngine = context.newEngine(ByteBufAllocator.DEFAULT);
        assertThat(sslEngine.getEnabledCipherSuites(), equalTo(sslSimpleBuilder.getCiphers()));
        assertThat(sslEngine.getEnabledProtocols(), equalTo(sslSimpleBuilder.getProtocols()));
        return sslEngine;
    }
```    
----

However if you don't accept the change, there a couple of nitpicks:
- the variable  `SslSimpleBuilder` should be lowercase started
- the `if`s are a `switch` statement, so using switch would grant that if future we don't miss any new enum value added to  `SslClientVerifyMode`

File: src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (andsel) [reply src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java:43]: :+1: maybe we can add an item in the task list in #14905 to track this and don't lose it

**Review by andsel — APPROVED** [review]
LGTM

**Pull request merged**